components/ProfilePage/SmartWalletContents.tsx

```diff
@@ -30,22 +38,41 @@ const SmartWalletContents = () => {
     getDNAByCre8orNumber()
   }, [getDNAByCre8orNumber])
 
-  const [, drop] = useDrop(() => ({
-    accept: ItemTypes.CRE8OR,
-    drop: () => {},
-    collect: (monitor) => ({
-      isOver: monitor.isOver(),
-      canDrop: monitor.canDrop(),
+  const [, drop] = useDrop(
+    () => ({
+      accept: ItemTypes.CRE8OR,
+      drop: async (item: any) => {
+        if (!hasSmartWallet || isHiddenEditable || !signer || !cre8orNumber) return
+
+        const smartWalletAddress = await getSmartWallet(cre8orNumber)
+
+        const response = await transferCre8orToSmartWallet(
+          item?.token.contractAddress,
+          address,
+          smartWalletAddress,
+          item?.token.tokenId,
+          signer,
+        )
+        if (!response?.err) {
```

Comment:
let's move this to a `useERC721Transfer` hook 🪝

Reply:
made hook 👍